### PR TITLE
Nextgen developments

### PR DESCRIPTION
**Pull request by wiz0u**
More dev-friendly library + using generated code for fast release

### CODE REVIEW

File: src/Telegram.Bot/Requests/Available methods/Manage Chat/RestrictChatMemberRequest.cs [hunk 1]
```diff
@@ -1,68 +1,51 @@
...
     /// <summary>
-    /// Date when restrictions will be lifted for the user, unix time. If user is restricted for
-    /// more than 366 days or less than 30 seconds from the current time, they are considered to
-    /// be restricted forever.
+    /// Date when restrictions will be lifted for the user, in UTC. If user is restricted for more than 366 days or less than 30 seconds from the current time, they are considered to be restricted forever
     /// </summary>
-    [JsonConverter(typeof(UnixDateTimeConverter))]
     [JsonInclude]
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [JsonConverter(typeof(BanTimeConverter))]
```
Comment (tuscen) [inline src/Telegram.Bot/Requests/Available methods/Manage Chat/RestrictChatMemberRequest.cs:38]: Why was the converter changes? BanTimeConverter was written for a different time if I remember it correctly

Comment (wiz0u) [reply src/Telegram.Bot/Requests/Available methods/Manage Chat/RestrictChatMemberRequest.cs:38]: generator of code sees "DateTime" "until_date" and considers it's a ban time.

now that BanTimeConverter just produce a 0 when DateTime is unspecified, which is fine for this method (means forever)

>Date when restrictions will be lifted for the user; Unix time. If user is restricted for more than 366 days or less than 30 seconds from the current time, they are considered to be restricted forever

File: src/Telegram.Bot/Types/ChatBackground.cs [hunk 2]
```diff
@@ -1,15 +0,0 @@
-﻿using JetBrains.Annotations;
-
-namespace Telegram.Bot.Types;
-
-/// <summary>
-/// This object represents a chat background
-/// </summary>
-[PublicAPI]
-public class ChatBackground
```
Comment (tuscen) [inline src/Telegram.Bot/Types/ChatBackground.cs:9]: Why was it removed?

Comment (wiz0u) [reply src/Telegram.Bot/Types/ChatBackground.cs:9]: moved into BackgroundType.cs

File: src/Telegram.Bot/Types/BusinessConnection.cs [hunk 3]
```diff
@@ -1,55 +1,44 @@
...
     [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
     [JsonConverter(typeof(UnixDateTimeConverter))]
     public DateTime Date { get; set; }
 
     /// <summary>
     /// <see langword="true"/>, if the bot can act on behalf of the business account in chats that were active in the last 24 hours
     /// </summary>
-    [JsonRequired]
-    [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
```
[the remark this answers is not in the document]
Comment (wiz0u) [reply src/Telegram.Bot/Types/BusinessConnection.cs:36]: As I warned beforehand, I changed a lot of bool? into bool to be more dev-friendly. In 99% of the cases, it's behaving the same (and I checked telegram-bot-api for that: when bool parameter is unspecified in the request, the default is false). The only documented exception is for anonymous polls which is true by default.

In this particular case anyway, BusinessConnection is never serialized, only deserialized.

File: src/Telegram.Bot/Types/Message.cs [hunk 4]
```diff
@@ -1,772 +1,652 @@
...
     public MessageEntity[]? Entities { get; set; }
 
     /// <summary>
-    /// Gets the entity values.
-    /// </summary>
-    /// <value>
-    /// The entity contents.
-    /// </value>
-    [JsonIgnore]
-    public IEnumerable<string>? EntityValues =>
```
Comment (tuscen) [inline src/Telegram.Bot/Types/Message.cs:254]: Why was it deleted?

Comment (wiz0u) [reply src/Telegram.Bot/Types/Message.cs:254]: I answered above. Extend.Types.cs

File: src/Telegram.Bot/Types/ReplyMarkups/InlineKeyboardMarkup.cs [hunk 5]
```diff
@@ -1,95 +1,27 @@
...
-        new(Array.Empty<InlineKeyboardButton[]>());
+    public InlineKeyboardMarkup(IEnumerable<IEnumerable<InlineKeyboardButton>> inlineKeyboard) => InlineKeyboard = inlineKeyboard;
 
     /// <summary>
-    /// Generate an inline keyboard markup with one button
+    /// Instantiates a new <see cref="InlineKeyboardMarkup"/>
     /// </summary>
-    /// <param name="button">Inline keyboard button</param>
-    [return: NotNullIfNotNull(nameof(button))]
-    public static implicit operator InlineKeyboardMarkup?(InlineKeyboardButton? button) =>
```
Comment (tuscen) [inline src/Telegram.Bot/Types/ReplyMarkups/InlineKeyboardMarkup.cs:69]: Why were implicit convertors removed?

Comment (wiz0u) [reply src/Telegram.Bot/Types/ReplyMarkups/InlineKeyboardMarkup.cs:69]: same, Extend.Types.cs

**Review by tuscen — APPROVED** [review]

**Pull request merged**